compiler/x/codegen/BinaryEvaluator.cpp

```diff
@@ -1223,7 +1223,19 @@ TR::Register *OMR::X86::TreeEvaluator::caddEvaluator(TR::Node *node, TR::CodeGen
    TR::MemoryReference *tempMR              = NULL;
    bool                 oursIsTheOnlyMemRef = true;
    TR::Compilation     *comp                = cg->comp();
+   
+   if (NEED_CC(node))
+      {    
+      TR_ASSERT(node->getOpCodeValue() == TR::cadd,
+                "CC computation not supported for this node %p with opcode %s\n", node, cg->comp()->getDebug()->getName(node->getOpCode()));
 
+      TR_X86BinaryCommutativeAnalyser  temp(cg);
+      // In caddEvaluator, if the second child is a const, the evaluator will change change those to 4 bytes lea.
```

Comment:
It's probably better to move this comments before "if (NEED_CC(node))" since that's where the logic starts. The second sentence could be merged with the first one "Need special handling when Eflags results are needed. Otherwise, the later part of caddEvaluator could optimize the instruction to a 4 bytes LEA or ADD4RegReg which only works when Eflags can be ignored."